### PR TITLE
chore(package): Bump dev deps

### PR DESCRIPTION
**Pull request by sudo-suhas**
**What**: Bump dev dependencies and other minor style changes.

**Why**: Good to have the deps updated and it also fixes a peer dependency warning.

**How**: Updated the package.json, fixed eslint errors, removed a duplicate test

**Checklist**:

- [ ] Documentation
- [ ] Tests
- [x] Ready to be merged 
- [x] Added myself to contributors table 

### Updated dev deps

|Package             |Current |Latest |
|--------------------|--------|-------|
|eslint              |3.19.0  |4.5.0  |
|eslint-plugin-babel |4.1.1   |4.1.2  |
|sinon               |1.17.7  |3.2.1  |

Useful links:
- sinon [migrating to 2.0](http://sinonjs.org/releases/v3.2.1/migrating-to-2.0/).
- sinon [migrating to 3.0](http://sinonjs.org/releases/v3.2.1/migrating-to-3.0/).
- eslint [migrating to 4.0](https://eslint.org/docs/user-guide/migrating-to-4.0.0).

This also fixes the following warning:

```
warning "eslint-config-kentcdodds@12.4.2" has incorrect peer dependency "eslint@>3".
```

P.S: If you don't mind me asking, why is sinon used in the project? Since jest has almost similar functionality inbuilt.

### CODE REVIEW

File: src/bin-utils/__tests__/parser.js [hunk 1]
```diff
@@ -214,20 +214,6 @@ test('if help is called with a script, it shows the help for that script', () =>
   expect(mockGetLogger.mock.info).toHaveBeenCalledTimes(1)
 })
 
-test('if help is called with a script, it shows the help for that script', () => {
```
Comment (kentcdodds) [inline src/bin-utils/__tests__/parser.js]: Why was this removed?

**Comment by kentcdodds** [comment]
Thanks for this! I'm using sinon because I haven't had time to remove it. This project started with ava I believe. Happy to accept a pull request for moving away from sinon!

File: src/bin-utils/__tests__/parser.js (hunk 1 above)
Comment (sudo-suhas) [reply src/bin-utils/__tests__/parser.js]: There was an eslint error(jest/no-identical-title) which drew my attention to it. The test is actually a repetition - https://github.com/kentcdodds/nps/blob/9d695d8feded0eb8654ec09c744257678bbb7cea/src/bin-utils/__tests__/parser.js#L203-L229

Comment (kentcdodds) [reply src/bin-utils/__tests__/parser.js]: Oh! Haha 😅 Thanks!
